File: C++/1496. Path Crossing.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isPathCrossing(string path)
    {
        set<string>hash;
        int x = 0, y = 0;
        hash.insert("0-0");
        for (int i = 0;i < path.size();++i)
        {
            if (path[i] == 'N')
                ++y;
            else if (path[i] == 'S')
                --y;
            else if (path[i] == 'E')
                ++x;
            else
                --x;

            string temp = to_string(x) + '-' + to_string(y);
            if (hash.find(temp) != hash.end())
                return true;
            else
                hash.insert(temp);
        }
        return false;

    }
};

File: Python/1496. Path Crossing.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        hash_set = set()
        x, y = 0, 0
        hash_set.add("0-0")

        for move in path:
            if move == 'N':
                y += 1
            elif move == 'S':
                y -= 1
            elif move == 'E':
                x += 1
            else:
                x -= 1

            temp = f"{x}-{y}"
            if temp in hash_set:
                return True
            else:
                hash_set.add(temp)

        return False
